dev/com.ibm.ws.webcontainer.servlet.3.0.factories/src/com/ibm/ws/webcontainer/osgi/webapp/internal/WebAppFactoryImpl.java

```diff
@@ -20,15 +23,26 @@
 import com.ibm.ws.webcontainer.osgi.webapp.WebAppFactory;
 import com.ibm.wsspi.injectionengine.ReferenceContext;
 
-@Component(service=WebAppFactory.class, property = { "service.vendor=IBM" })
-public class WebAppFactoryImpl implements WebAppFactory {
+import io.openliberty.checkpoint.spi.CheckpointHook;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+
+@Component(service= {WebAppFactory.class, CheckpointHook.class}, property = { "service.vendor=IBM" })
+public class WebAppFactoryImpl implements WebAppFactory, CheckpointHook {
+	private static CheckpointPhase _checkpointPhase;
```

Comment:
Are these supposed to be static in each of the classes?

Reply:
No, make these non-static private finals.